### PR TITLE
[WIP] Implement onboarding workflow

### PR DESCRIPTION
**Pull request by hebasto**
This PR implements **Bosch-0**'s design:

Screenshot from 2021-10-02 21-42-07

Based on #42, and only the last commit belongs to this PR.

### CODE REVIEW

**Comment by hebasto** [comment]
cc @Bosch-0

File: src/qml/pages/onboarding/Onboarding00Welcome.qml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+
+Column {
+    width: appTitle.width
+    spacing: 16
+
+    Image {
+        id: appLogo
```
Comment (promag) [inline src/qml/pages/onboarding/Onboarding00Welcome.qml:15]: 21c969272e718e63d3de00a9fe9e0f702eb4fd9f

Let's avoid unnecessary names (`id:`).

I also think these should be snake_case to distinguish from property names.

File: src/qml/pages/onboarding/Onboarding00Welcome.qml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        anchors.horizontalCenter: parent.horizontalCenter
+        source: "image://images/app"
+        sourceSize.width: 128
+        sourceSize.height: 128
+    }
+
+    Text {
```
Comment (promag) [inline src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: 21c969272e718e63d3de00a9fe9e0f702eb4fd9f

Should use `Label` instead.

File: src/qml/pages/onboarding/Onboarding00Welcome.qml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+// The Onboarding00Welcome page.
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+
+Column {
```
Comment (promag) [inline src/qml/pages/onboarding/Onboarding00Welcome.qml:10]: 21c969272e718e63d3de00a9fe9e0f702eb4fd9f

We should extend `Pane`, something like:

```qml
Pane {
    contentItem: ColumnLayout {
        // ...
    }
}
```

**Review by promag — COMMENTED** [review]
Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43. See 

https://user-images.githubusercontent.com/3534524/135757737-be096e27-96db-40cc-840c-a9f236590ebe.mov

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 1 above)
Comment (hebasto) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:15]: > Let's avoid unnecessary names (`id:`).

Agree.

> I also think these should be snake_case to distinguish from property names.

I'd prefer to follow QML style provided by Qt docs. It guarantees more consistency in long run with more contributors.

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 2 above)
Comment (hebasto) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: `Label` extends `Text`. Which `Label`'s properties we are interested in?

**Comment by jarolrod** [comment]
> Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43. See
> 
>  Screen.Recording.2021-10-03.at.15.14.27.mov

@Bosch-0 @GBKS In regards to this video: Shouldn't the windows have a minimum size, and minimum amount of padding?

**Comment by promag** [comment]
> > Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43. See
> > Screen.Recording.2021-10-03.at.15.14.27.mov
> 
> @Bosch-0 @GBKS In regards to this video: Shouldn't the windows have a minimum size, and minimum amount of padding?

Definitely, this is just to show that layouts is superior to anchors.

**Comment by jarolrod** [comment]
@promag oh right, sorry!

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 2 above)
Comment (promag) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: I’d say background and palette.

**Comment by hebasto** [comment]
@promag
> Pushed my suggestions to https://github.com/promag/gui-qml/tree/qml-43.

`wrapMode: Text.WordWrap` does not work as expected.

**Comment by hebasto** [comment]
Updated 21c969272e718e63d3de00a9fe9e0f702eb4fd9f -> 7fc49d75c6da0a1fe700458e4130ad18df3a04ae ([pr43.01](https://github.com/hebasto/gui-qml/commits/pr43.01) -> [pr43.02](https://github.com/hebasto/gui-qml/commits/pr43.02)):

- rebased
 - addressed some of @promag's comments

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 2 above)
Comment (hebasto) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml]: Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/43#issuecomment-933918510).

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 3 above)
Comment (hebasto) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:10]: > We should extend `Pane`...

Why? We already have `ApplicationWindow` container.

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 1 above)
Comment (hebasto) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml]: > Let's avoid unnecessary names (`id:`).

[Done](https://github.com/bitcoin-core/gui-qml/pull/43#issuecomment-933918510).

**Comment by hebasto** [comment]
Updated 7fc49d75c6da0a1fe700458e4130ad18df3a04ae -> 6a19c31561c658c961f593faef5d151c1e68ef1d ([pr43.02](https://github.com/hebasto/gui-qml/commits/pr43.02) -> [pr43.04](https://github.com/hebasto/gui-qml/commits/pr43.04)):

- implemented @promag's suggestions to use `ColumnLayout` and `Pane`

Added commits are supposed to be squashed before merging.

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 3 above)
Comment (hebasto) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml]: [Updated](https://github.com/bitcoin-core/gui-qml/pull/43#issuecomment-933995134).

File: src/qml/pages/onboarding/Onboarding00Welcome.qml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+// The Onboarding00Welcome page.
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+
+Pane {
+    background: Rectangle {
```
Comment (promag) [inline src/qml/pages/onboarding/Onboarding00Welcome.qml:12]: Not needed? The window background is already black. To improve rendering time make `background: null`

Comment (promag) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:12]: Documentation for null or for rendering improvement?

File: src/qml/pages/onboarding/Onboarding00Welcome.qml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        color: "black"
+    }
+
+    ColumnLayout {
+        anchors.centerIn: parent
+        spacing: 16
+
+        Image {
+            Layout.alignment: Qt.AlignCenter
+            source: "image://images/app"
```
Comment (jarolrod) [inline src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: making a note that @bosch-0 has a proposal for a new icon. It needs to be discussed at greater length with the community; if there is consensus we should use it: https://github.com/bitcoin-core/gui/pull/199

File: src/qml/pages/onboarding/Onboarding00Welcome.qml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            id: appTitle
+            Layout.alignment: Qt.AlignCenter
+            text: "Bitcoin Core TnG"
+            font.pointSize: 36
+            color: "white"
+        }
+
+        Label {
+            Layout.maximumWidth: appTitle.width
+            text: "Be part of the Bitcoin network."
```
Comment (jarolrod) [inline src/qml/pages/onboarding/Onboarding00Welcome.qml:37]: wrap in `qsTr` in preparation for translation support?
```suggestion
            text: qsTr("Be part of the Bitcoin network.")
```

File: src/qml/pages/onboarding/Onboarding00Welcome.qml (hunk 5 above)
Comment (Bosch-0) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: It's not just a new icon, there is also some changes to the watermark and thus overall logo. See this file for details: https://www.figma.com/file/c1V7b23n0LqRbVJlUkE1mn/Bitcoin-Core-App-Bosch?node-id=1850%3A44

Comment (GBKS) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: The new proposed icon is visually not centered in the circle. Can we address that, please?

Also, there are two oranges in the Figma file, Bitcoin Core orange and Bitcoin orange. They are slightly different, which one is the "right one"?

Comment (Bosch-0) [reply src/qml/pages/onboarding/Onboarding00Welcome.qml:22]: I updated that awhile ago, looks visually centered to me?

image


The Bitcoin Core orange is the primary orange, the Bitcoin orange is just there to illustrate the orange used by the Bitcoin logo - ill remove it from the stylesheet.

**Comment by hebasto** [comment]
Closing in favor of #124.

**Pull request closed without merging**